src/cargo/util/errors.rs

```diff
@@ -286,6 +287,37 @@ impl From<Box<CargoError>> for CliError {
 }
 
 // =============================================================================
+// NetworkError trait
+
+pub trait NetworkError: CargoError {
+    fn maybe_spurious(&self) -> bool;
+}
+
+impl NetworkError for git2::Error {
+    fn maybe_spurious(&self) -> bool {
+        match self.class() {
+            git2::ErrorClass::Net |
+            git2::ErrorClass::Os => true,
+            _ => false
+        }
+    }
+}
+impl NetworkError for curl::ErrCode {
+    fn maybe_spurious(&self) -> bool {
+        println!("{}", self.code() as i32);
+        match self.code()  {
+            curl_sys::CURLcode::CURLE_COULDNT_CONNECT |
+            curl_sys::CURLcode::CURLE_COULDNT_RESOLVE_PROXY |
+            curl_sys::CURLcode::CURLE_COULDNT_RESOLVE_HOST |
+            curl_sys::CURLcode::CURLE_OPERATION_TIMEDOUT |
```

Comment:
It looks like curl retries (at least once) on CURLE_OPERATION_TIMEDOUT  https://github.com/curl/curl/blob/b9728bca549709a26a5228f1d44f7488dd26811d/src/tool_operate.c#L1449

I do not have any data around which errors people are seeing. It looks like, after a quick search in the curl code, that CURLE_OPERATION_TIMEDOUT covers a lot of different case. After this is merge the list can be changed easy. In this case I believe catching more is better only because if this error code turns out to be bad the work around would be to configure the setting. If you release cargo without this and it turns out that it is the core failure case for curl everyone who is affected need to wait for a new release cycle. 

Let me know if you would still like to remove it.